client/offchain/src/api.rs

```diff
@@ -180,6 +182,33 @@ impl<Storage: OffchainStorage> OffchainExt for Api<Storage> {
 	) -> Result<usize, HttpError> {
 		self.http.response_read_body(request_id, buffer, deadline)
 	}
+
+	fn get_node_public_key(&mut self) -> Result<NodePublicKey, ()> {
+		let public_key = self.network_state.local_public_key();
+		match public_key {
+			identity::PublicKey::Ed25519(public) =>
+				Ok(public.encode().into()),
+			_ => Err(()),
```

Comment:
What about other cases? We don't support them?

Reply:
Yeah, the network layer now only use ed25519, likely others will also be supported, don't know when though.